Lib/getpass.py

```diff
@@ -151,6 +176,42 @@ def _raw_input(prompt="", stream=None, input=None):
     return line
 
 
+def _input_with_echochar(prompt, stream, input, echochar):
```

Comment:
rather than duplicating so much of _raw_input's logic, I suggest making a _readline_with_echochar() function and just conditionally calling that vs the existing input.readline() call within _raw_input().

Reply:
sounds reasonable, please check my diff